middleware/database/database.go

```diff
@@ -0,0 +1,95 @@
+package database
+
+import (
+	"database/sql"
+	"database/sql/driver"
+	"os"
+
+	sqltrace "gopkg.in/DataDog/dd-trace-go.v1/contrib/database/sql"
+)
+
+// RegisterDriverAndOpen registers the selected driver with the datadog-lib, and opens a connection to the database using the dsn.
+func RegisterDriverAndOpen(driverName string, driver driver.Driver, dsn string, options ...Option) (*sql.DB, error) {
+	cfg := defaults()
+	for _, opt := range options {
+		opt(cfg)
+	}
+	opts := make([]sqltrace.Option, 0, 4)
```

Comment:
If you are trying to fix the capacity and trying to minimize re-allocations, should it not be?

```
opts := make([]sqltrace.Option, 0, 2 + len(cfg.tags) + len(ignoredQueryTypes))
```

Reply:
Actually, this is not that important. And if you deem that this change is necessary, then you can include it in a separate PR.

Reply:
A remnant from a previous iteration :) 
I'll create a separate PR for improving it, but this func wont be in the hot-loop, so it won't matter much :)